edb/pgsql/compiler/__init__.py

```diff
@@ -89,28 +92,31 @@ def compile_ir_to_sql_tree(
 
 
 def compile_ir_to_sql(
-        ir_expr: irast.Base, *,
-        output_format: Optional[OutputFormat]=None,
-        ignore_shapes: bool=False,
-        explicit_top_cast: Optional[irast.TypeRef]=None,
-        use_named_params: bool=False,
-        expected_cardinality_one: bool=False,
-        pretty: bool=True) -> Tuple[str, Dict[str, int]]:
+    ir_expr: irast.Base, *,
+    output_format: Optional[OutputFormat]=None,
+    ignore_shapes: bool=False,
+    explicit_top_cast: Optional[irast.TypeRef]=None,
+    use_named_params: bool=False,
+    expected_cardinality_one: bool=False,
+    pretty: bool=True,
+    argmap: Dict[str, int]=None,
```

Comment:
I'm afraid this makes the API harder to use.  The `argmap` doesn't *appear* as a required argument, whereas now it has to be carefully constructed by the caller or the compilation will crash.  It seems like building `argmap` is a concern separate from filtering out unused params, can we keep it inside `compile_ir_to_sql`?